subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryWalkerTest.groovy

```diff
@@ -16,78 +16,22 @@
 
 package org.gradle.api.internal.file.collections
 
-import com.google.common.base.Charsets
-import org.gradle.api.JavaVersion
-import org.gradle.api.file.FileVisitDetails
 import org.gradle.api.file.FileVisitor
 import org.gradle.api.internal.file.TestFiles
-import org.gradle.api.internal.file.collections.jdk7.Jdk7DirectoryWalker
 import org.gradle.api.tasks.util.PatternSet
-import org.gradle.internal.Factory
 import org.gradle.test.fixtures.file.TestFile
-import org.gradle.util.Requires
 import org.gradle.util.UsesNativeServices
-import spock.lang.Issue
 
-import java.nio.charset.Charset
 import java.util.concurrent.atomic.AtomicInteger
 
 @UsesNativeServices
 class DirectoryWalkerTest extends AbstractDirectoryWalkerTest<DirectoryWalker> {
     @Override
     protected List<DirectoryWalker> getWalkers() {
-        return [new DefaultDirectoryWalker(), new Jdk7DirectoryWalker(), new ReproducibleDirectoryWalker()]
-    }
-
-    // java.nio2 cannot access files with unicode characters when using single-byte non-unicode platform encoding
-    // bug seems to show up only on JDK7 when file.encoding != sun.jnu.encoding
-    @Issue("GRADLE-2181")
-    @Requires(adhoc = { JavaVersion.current().isJava7() && (System.getProperty("sun.jnu.encoding") == null || Charset.forName(System.getProperty("sun.jnu.encoding")).contains(Charsets.UTF_8)) })
-    def "check that JDK7 walker gets picked with Unicode encoding as default"() {
-        setup:
-        System.setProperty("file.encoding", fileEncoding)
-        Charset.defaultCharset = null
-        def directoryWalkerFactory = TestFiles.directoryFileTreeFactory().create(tmpDir.createDir("root")).directoryWalkerFactory
-        directoryWalkerFactory.reset()
-        expect:
-        directoryWalkerFactory.create().class.simpleName == expectedClassName
-        where:
-        fileEncoding | expectedClassName
-        "UTF-8"      | "Jdk7DirectoryWalker"
-        "UTF-16be" | "Jdk7DirectoryWalker"
-        "UTF-16le" | "Jdk7DirectoryWalker"
-        "UTF-16"   | "Jdk7DirectoryWalker"
-        "ISO-8859-1" | "DefaultDirectoryWalker"
-    }
-
-    def "both DirectoryWalker implementations return same set of files and attributes"() {
```

Comment:
We need to keep this test now between `ReproducibleDirectoryWalker` and `DefaultDirectoryWalker`, right?

Reply:
reverted and changed, thanks for the insight.